components/protected-route/ProtectedRoute.tsx

```diff
@@ -0,0 +1,30 @@
+import { useAuthContext } from "@/context/AuthContext";
+import { useRouter } from "next/router";
+import { FC } from "react";
+
+
+
+const WithProtectedRoute = <P extends object>(
+    WrappedComponent: FC<P>
+): FC<P> => {
+    const Wrapper = (props: P) => {
+        const router = useRouter();
+        const { user, isFirstLogin } = useAuthContext();
+        
+        if (!user) {
+            router.push("./signin");
```

Comment:
Also sign in are two words so maybe rename this and the folder to sign-in

Reply:
eehhh okay